main_comment.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/gin-gonic/gin"
+	"github.com/google/go-github/v28/github"
+
+	clientgithub "github.com/mendersoftware/integration-test-runner/client/github"
+)
+
+func processGitHubComment(ctx *gin.Context, comment *github.IssueCommentEvent, githubClient clientgithub.Client, conf *config) error {
+	log := getCustomLoggerFromContext(ctx)
+
+	// process created actions only, ignore the others
+	action := comment.GetAction()
+	if action != "created" {
+		log.Infof("Ignoring action %s on comment", action)
+		return nil
+	}
+
+	// accept commands only from organization members
+	if member := githubClient.IsOrganizationMember(ctx, githubOrganization, comment.Sender.GetLogin()); !member {
+		log.Warnf("%s commented, but he/she is not a member of our organization, ignoring", comment.Sender.GetLogin())
+		return nil
+	}
+
+	// filter comments mentioning the bot
+	body := comment.Comment.GetBody()
+	if !strings.Contains(body, "@"+githubBotName) {
+		log.Infof("ignoring comment not mentioning me")
+		return nil
+	}
+
+	// extract the command and check it is valid
+	command := ""
+	if strings.Contains(body, commandStartPipeline) {
```

Comment:
Question:
Could not all of this be a part of the switch statement itself? :thinking:

Reply:
Done